### DISCUSSION TITLE
update search params and skip loader reload

### DISCUSSION THREAD

**Discussion by apalumbo** [remix-run/remix#4987] [Show and tell / tips]
We are migrating from nextjs to remix, in a page we are using the an option of nextjs router (shallowRender) to change the url and skip data fetching.
we only need to avoid to re-invoke the loader when pushing the updated current page (is an infinite scroll that use an external search engine on both client and server side) or at least detect in the loader code the fetch and avoid to call the search engine 2 times (one on the server and one on the client)
I'm already aware of this issue https://github.com/remix-run/react-router/issues/8058, I think that the use case is different
Thankyou

**Comment by apalumbo** [comment]
After reading a lot of code ;-) I found the remix way to do it. we will use the `useSearchParams` hook to set the new search params 
to skip the loader reload you could export the unstable_shouldReload function that must return `false` id the reload must be skipped `true` otherwise.

```
export const unstable_shouldReload: ShouldReloadFunction = ({
    // same params that go to "loader" and "action"
    params,

    // a possible form submission that caused this to be reloaded
    submission,

    // the next URL being used to render this page
    url,

    // the previous URL used to render this page
    prevUrl,
  }) => {
    return true; // you have to handle here the logic to return false if you want to avoid the loader reload
 }
```

**Reply by aniravi24** [reply]
You can also use the [useNavigate](https://reactrouter.com/en/6.6.1/hooks/use-navigate) hook if you generate a query string to pass to it.

And yes the current API is [unstable_shouldReload](https://remix.run/docs/en/v1/route/should-reload#unstable_shouldreload), but it should be stabilized in the [next release](https://github.com/remix-run/remix/releases/tag/remix%401.10.0-pre.5)!
